assets/src/scripts/models.js

```diff
@@ -42,33 +42,126 @@ export function getTimeseries({sites, params=['00060'], startDate=null, endDate=
         .then((response) => {
-            let data = JSON.parse(response);
-            return data.value.timeSeries.map(series => {
-                    let startDate = new Date(series.values[0].value[0].dateTime);
-                    let endDate = new Date(
-                        series.values[0].value.slice(-1)[0].dateTime);
-                    return {
-                        code: series.variable.variableCode[0].value,
-                        variableName: series.variable.variableName,
-                        variableDescription: series.variable.variableDescription,
-                        seriesStartDate: startDate,
-                        seriesEndDate: endDate,
-                        values: series.values[0].value.map(value => {
-                            let date = new Date(value.dateTime);
-                            return {
-                                time: date,
-                                value: parseFloat(value.value),
-                                label: `${formatTime(
-                                    date)}\n${value.value} ${series.variable.unit.unitCode}`
-                            };
-                        })
-                    };
+                let data = JSON.parse(response);
+                return data.value.timeSeries.map(series => {
+                        let startDate = new Date(series.values[0].value[0].dateTime);
+                        let endDate = new Date(
+                            series.values[0].value.slice(-1)[0].dateTime);
+                        return {
+                            code: series.variable.variableCode[0].value,
+                            variableName: series.variable.variableName,
+                            variableDescription: series.variable.variableDescription,
+                            seriesStartDate: startDate,
+                            seriesEndDate: endDate,
+                            values: series.values[0].value.map(value => {
+                                let date = new Date(value.dateTime);
+                                return {
+                                    time: date,
+                                    value: parseFloat(value.value),
+                                    label: `${formatTime(
+                                        date)}\n${value.value} ${series.variable.unit.unitCode}`
+                                };
+                            })
+                        };
+                    }
+                );
+            },
+            (error) => {
+                return error;
+            });
+}
+export function getSiteStatistics({sites, statType, params=['00060']}) {
+    let url = `${SERVICE_ROOT}/stat/?format=rdb&sites=${sites.join(',')}&statReportType=daily&statTypeCd=${statType}&parameterCd=${params.join(',')}`;
+    return get(url);
+}
+
+
+/**
+ * Function to parse RDB to Objects
+ */
+export function parseRDB(rdbData) {
+    let rdbLines = rdbData.split('\n');
```

Comment:
I suppose there's not a great way to do this in the browser, without pulling in a new dependency - but it might be worth researching if we can iterate over this without splitting and creating the intermediate array.

Reply:
I'll take a look into that.